Add Stencil

arjunyel opened the pull request:
https://github.com/webcomponents/custom-elements-everywhere/issues/13

arjunyel commented:
Locally when I run `npm run build` I get the error `Error: Missing results.json` for Stencil and Dojo but I have no idea whats wrong :(

robdodson commented:
Hi @arjunyel, I'll try to take a look at this today.

robdodson commented on docs/index.js:
```diff
@@ -33,6 +33,7 @@ const libraryMap = {
   react: "React",
   riot: "Riot.js",
   skate: "Skate w/ Preact",
+  stencil: 'Stencil',
```
```suggestion
  stencil: "Stencil",
```

robdodson commented on libraries/stencil/.gitignore:
```diff
@@ -0,0 +1,26 @@
+dist/
```
I don't think any of the other libraries include their own `.gitignore`. Is this needed?

arjunyel replied:
Since Stencil is a compiler we do need to ignore the ouput/cache folders or it will bloat the repository, however I cleaned the gitignore so its just stencil specific stuff is that okay?

robdodson commented on libraries/stencil/package.json:
```diff
@@ -0,0 +1,28 @@
+  "types": "dist/types/components.d.ts",
+  "collection": "dist/collection/collection-manifest.json",
+  "collection:main": "dist/collection/index.js",
+  "unpkg": "dist/stencil-elements-everywhere/stencil-elements-everywhere.js",
```
do we need any of these fields?

robdodson commented on libraries/stencil/package.json:
```diff
@@ -0,0 +1,28 @@
+  "devDependencies": {
+    "@babel/core": "7.6.2",
+    "babel-loader": "8.0.6",
+    "@babel/preset-env": "7.6.2",
```
I think the other libraries are on:

```
"babel-core": "6.26.3",
"babel-loader": "7.1.5",
"babel-preset-es2015": "6.24.1",
```

If possible, I'd suggest using those dependencies.

robdodson commented on libraries/stencil/package.json:
```diff
@@ -0,0 +1,28 @@
+    "karma-firefox-launcher": "1.2.0",
+    "karma-sourcemap-loader": "0.3.7",
+    "karma-webpack": "4.0.2",
+    "webpack": "4.41.0",
```
You might want to use these dependency versions as well:

```
"cross-env": "5.2.1",
"karma": "4.3.0",
"karma-chrome-launcher": "2.2.0",
"karma-firefox-launcher": "1.2.0",
"karma-sourcemap-loader": "0.3.7",
"karma-webpack": "3.0.5",
```

robdodson commented on libraries/stencil/src/components/component-with-children-rerender/readme.md:
```diff
@@ -0,0 +1,17 @@
+# component-with-children-rerender
```
is it possible to not generate these files?

robdodson commented on libraries/stencil/package.json:
```diff
@@ -0,0 +1,28 @@
+    "karma-sourcemap-loader": "0.3.7",
+    "karma-webpack": "4.0.2",
+    "webpack": "4.41.0",
+    "@stencil/core": "1.5.0"
```
The build is failing because this is in devDependencies and not dependencies. If you move it to dependencies it should work.

arjunyel commented:
Thanks for the help Rob! I believe besides the .gitignore I've addressed all the feedback. BTW I watched all your polycasts back in the day, they were a huge help to me!

robdodson commented:
Rad, thank you @arjunyel! I know this was a lot of work and I really appreciate it :D
